### PR TITLE
GUI-1773, GUI-1861 handle templates with no parameters and fixed vpc sample template

### PR DESCRIPTION
**Pull request by dkavanagh**
handle templates with no parameters and fixed vpc sample template

https://eucalyptus.atlassian.net/browse/GUI-1773
https://eucalyptus.atlassian.net/browse/GUI-1861
https://eucalyptus.atlassian.net/browse/GUI-1865
https://eucalyptus.atlassian.net/browse/GUI-1878

### CODE REVIEW

**Comment by aliceh** [comment]
retest this, please

**Comment by dkavanagh** [comment]
It looks like the last CI run passed: http://10.111.4.1/job/user-console-regression-sequence/14283/

File: eucaconsole/templates/stacks/stack_wizard.pt [hunk 1]
```diff
@@ -98,10 +98,13 @@
                             <div class="row">
                                 <div class="small-3 columns">&nbsp;</div>
                                 <div class="small-9 columns field inline">
-                                    <a id="visit-step-2" class="button small round" ng-click="visitNextStep(2, $event)" ng-disabled="isNotValid">
+                                    <button type="submit" class="button" id="stack-wizard-btn-step2" ng-disabled="isNotValid" ng-show="!isNotValid && !loading && parameters.length === 0">
```
Comment (kamalgill) [inline eucaconsole/templates/stacks/stack_wizard.pt]: @dkavanagh: The double ampersands need to be escaped here for XHTML

File: eucaconsole/templates/stacks/stack_wizard.pt [hunk 2]
```diff
@@ -31,7 +31,7 @@
                                 <span class="cir">1</span> <b i18n:translate="">General</b>
                             </a>
                         </dd>
-                        <dd ng-class="{disabled: step1Invalid}">
+                        <dd ng-class="{disabled: step1Invalid}" ng-show="isNotValid || loading || parameters.length > 0">
```
Comment (kamalgill) [inline eucaconsole/templates/stacks/stack_wizard.pt]: @dkavanagh: The '>' in `parameters.length > 0` needs to be entity-escaped

File: eucaconsole/views/stacks.py [hunk 3]
```diff
@@ -412,7 +414,12 @@ def stack_template_parse(self):
             except ValueError as json_err:
                 raise JSONError(message=_(u'Invalid JSON File ({0})').format(json_err.message), status=400)
             except HTTPError as http_err:
-                raise JSONError(message=_(u'Cannot read URL ({0})').format(http_err.reason), status=400)
+                raise JSONError(message=_(u"""
+                    Cannot read URL ({0}) If this URL is for an S3 object, be sure 
+                    that either the object has public read permissions or that the 
+                    URL is signed with authentication information.
+                 """).format(http_err.reason), status=400)
+
```
Comment (kamalgill) [inline eucaconsole/views/stacks.py]: @dkavanagh : This line needs to be removed for PEP8 compliance

File: eucaconsole/templates/stacks/stack_wizard.pt [hunk 4]
```diff
@@ -98,10 +98,13 @@
...
                                 <div class="small-9 columns field inline">
-                                    <a id="visit-step-2" class="button small round" ng-click="visitNextStep(2, $event)" ng-disabled="isNotValid">
+                                    <button type="submit" class="button" id="stack-wizard-btn-step2" ng-disabled="isNotValid" ng-show="!isNotValid && !loading && parameters.length === 0">
+                                        <span i18n:translate="">Create Stack</span>
+                                    </button>
+                                    <a id="visit-step-2" class="button small round" ng-click="visitNextStep(2, $event)" ng-show="isNotValid || loading || parameters.length > 0" ng-disabled="isNotValid || !parameters">
                                         <span i18n:translate="">Next</span>
-                                        <a class="cancel-link" href="${request.route_path('stacks')}">Cancel</a>
                                     </a>
+                                    <a class="cancel-link" href="${request.route_path('stacks')}">Cancel</a>
```
Comment (kamalgill) [inline eucaconsole/templates/stacks/stack_wizard.pt]: @dkavanagh: We're missing an i18n:translate attribute here.  Also, please open stack_wizard.pt in PyCharm to highlight the numerous "invalid HTML" violations (e.g. select elements cannot contain a 'value' attribute, which should be only added to the option elements inside a select element).

Comment (dkavanagh) [inline eucaconsole/templates/stacks/stack_wizard.pt]: I requested a pycharm license week before last, so hopefully, I'll be able to run it again soon.

**Comment by JMoLo** [comment]
Just FYI, PyCharm licenses have been ordered. May take a little longer than usual due to separation-related system shutdowns, but they were trying to push them through.

**Comment by dkavanagh** [comment]
I'm going to merge this since other stack-related fixes are going in and merging will be simpler for those w/ this already in master. If other issues exist, I'll fix after I get my license.

**Pull request merged**